File: backend/src/main/java/com/nanmuli/blog/domain/skill/Skill.java
package com.nanmuli.blog.domain.skill;

import com.nanmuli.blog.shared.domain.BaseAggregateRoot;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
public class Skill extends BaseAggregateRoot<Long> {
    private static final long serialVersionUID = 1L;

    private String name;
    private String category;
    private Integer proficiency;
    private String icon;
    private String color;
    private String description;
    private Integer sort;
    private Integer status;
}

File: frontend/src/types/skill.ts
// 注意：ID 使用 string 类型，因为 JavaScript number 无法精确表示 64 位整数
export interface Skill {
  id: string
  name: string
  category: 'language' | 'framework' | 'tool' | 'other'
  proficiency: number
  icon: string
  color: string
  description: string
  sort: number
  status: number
  createTime: string
  updateTime: string
}
